### PR TITLE
BUGZ-730: Fix deferred rendering on macbook air

### PR DESCRIPTION
**Pull request by SamGondelman**
https://highfidelity.atlassian.net/browse/BUGZ-730

### CODE REVIEW

File: libraries/render-utils/src/DeferredBufferWrite.slh [hunk 1]
```diff
@@ -69,7 +67,8 @@ void packDeferredFragmentTranslucent(vec3 normal, float alpha, vec3 albedo, floa
     }
     _fragColor0 = vec4(albedo.rgb, alpha);
     _fragColor1 = vec4(packNormal(normal), clamp(roughness, 0.0, 1.0));
-
+    _fragColor2 = vec4(vec3(0.0), 1.0);
+    _fragColor3 = vec4(0.0);
```
Comment (SamGondelman) [inline libraries/render-utils/src/DeferredBufferWrite.slh]: Not the fix but not having these have caused us issues in the past.

File: libraries/render-utils/src/DeferredLightingEffect.cpp [hunk 2]
```diff
@@ -41,33 +41,17 @@ namespace gr {
...
 using namespace render;
 
-struct LightLocations {
-    bool shadowTransform{ false };
-    void initialize(const gpu::ShaderPointer& program) {
-        shadowTransform = program->getReflection().validUniformBuffer(ru::Buffer::ShadowParams);
-    }
-};
-
-static void loadLightProgram(int programId, bool lightVolume, gpu::PipelinePointer& program, LightLocationsPtr& locations);
```
Comment (SamGondelman) [inline libraries/render-utils/src/DeferredLightingEffect.cpp]: Not the fix but the LightLocations stuff is unused.

File: libraries/render-utils/src/deferred_light.slv [hunk 3]
```diff
@@ -18,15 +18,15 @@ layout(location=RENDER_UTILS_ATTR_TEXCOORD01) out vec4 _texCoord01;
 
 void main(void) {
     const float depth = 1.0;
-    const vec4 UNIT_QUAD[4] = vec4[4](
+    const mat4 UNIT_QUAD = mat4(
```
Comment (SamGondelman) [inline libraries/render-utils/src/deferred_light.slv]: This is the fix.  /shrug
(without it, the slf gets the incorrect texCoords)

Comment (SamGondelman) [reply libraries/render-utils/src/deferred_light.slv]: There are other places where we use this pattern, but they seem to work.  I don't know what's different here.

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20030/HighFidelity-Beta-PR15796-3ff8160.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20030/HighFidelity-Beta-Interface-PR15796-3ff8160.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20030/HighFidelity-Beta-PR15796-3ff8160.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20030/HighFidelity-Beta-Interface-PR15796-3ff8160.exe

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20030/HighFidelity-Beta-PR15796-3ff8160.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20030/HighFidelity-Quest-Beta-PR15796-3ff8160.apk)

**Pull request merged**